docs/outlook/spam-reporting.md

```diff
@@ -0,0 +1,230 @@
+</VersionOverrides>
+```
+
+## Implement the event handler
+
+When your add-in is used to report a message, it generates a [SpamReporting](/javascript/api/office/office.eventtype?view=outlook-js-preview&preserve-view=true) event, which is then processed by the event handler in the JavaScript file of your add-in. To map the name of the event handler you specified in the **\<FunctionName\>** element of your manifest to its JavaScript counterpart, you must call [Office.actions.associate](/javascript/api/office/office.actions#office-office-actions-associate-member) in your code.
+
+Your event handler is responsible for processing the reported message, such as forwarding a copy of the message to an internal system for further investigation. To efficiently send a copy of the reported message, call the [getAsFileAsync](/javascript/api/outlook/office.messageread?view=outlook-js-preview&preserve-view=true#outlook-office-messageread-getasfileasync-member(1)) method in your event handler. This gets the Base64 encoding of a message, which you can then forward to your internal system.
+
+Once the event handler has completed processing the message, it must call the [event.completed](/javascript/api/office/office.addincommands.event#office-office-addincommands-event-completed-member(1)) method. In addition to signaling to the add-in that the spam-reporting event has been processed, `event.completed` can also be used to customize a post-processing dialog to show to the user or perform additional operations on the message, such as deleting it from the inbox. For a list of properties you can include in a JSON object to pass as a parameter to the `event.completed` method, see [Office.AddinCommands.EventCompletedOptions](/javascript/api/office/office.addincommands.eventcompletedoptions).
+
+> [!NOTE]
+> Code added after the `event.completed` call isn't guaranteed to run.
+
+The following is an example of a spam-reporting event handler.
+
+```javascript
+// Handles the SpamReporting event to process a reported message.
+function onSpamReport(event) {
+  // Gets the Base64 encoding of a reported message.
+  Office.context.mailbox.item.getAsFileAsync({ asyncContext: event }, (asyncResult) => {
+    if (asyncResult.status === Office.AsyncResultStatus.Failed) {
+      console.log(`Error encountered during message processing: ${asyncResult.error.message}`);
+      return;
+    }
+
+    // Run additional processing operations here.
+
+    /**
+     * Signals that the spam-reporting event has completed processing.
+     * It then moves the reported message to the Junk Email folder of the mailbox, then
+     * shows a post-processing dialog to the user. If an error occurs while the message
+     * is being processed, the `onErrorDeleteItem` property determines whether the message
+     * will be deleted.
+     */
+    const event = asyncResult.asyncContext;
+    event.completed({
+      onErrorDeleteItem: true,
+      postProcessingAction: moveToSpamFolder,
```

Comment:
I think this is an error moveToSpamFolder needs to be a string: 

postProcessingAction: "moveToSpamFolder" 

(at least until we fix the enum task)